### DISCUSSION TITLE
Compiler Fehler durch fehlende Dateiendung .js

### DISCUSSION THREAD

**Discussion by SLS-BLN** [neuefische/questions#54] [Web]
Nach einem PR ohne Änderungen und einem Merge bekomme ich im Browser eine seltsame Fehlermeldung. Wer kann mir eine Erklärung dafür liefern?

**Kurzes Udpate**
Der Fehler ist nicht reproduzierbar - Danke für die Unterstützung an @actyralf (Coach Ralf)  
Die Seite an sich läuft auch - d.h. wer zufällig die Lösung weiß: super; wenn nicht: auch kein Beinbruch! 

Das ist die Fehlermeldung:

<img width="800" alt="Screenshot 2022-08-10 at 10 38 39" src="https://user-images.githubusercontent.com/99472170/183855971-b0a703c7-dec1-464d-a3f9-3d7497b50ae6.png">

Vorher war alles in Ordnung. Nach ein paar Tests habe ich den "Schuldigen" gefunden. Es ist diese Zeile hier:

```
import {tagColors} from './BlogTagData';
```

Wenn ich jetzt eine kleine Änderung vornehme, funktioniert alles wieder - d.h. wenn ich hinten ein .js an den Name hänge.

```
import {tagColors} from './BlogTagData.js';
```

Der Fehler hat mich 45 min gekostet. Die Frage ist: warum? wie kann ich das in Zukunft vermeiden?
Hier ist noch der Link zu meinem [Capstone Project](https://github.com/SLS-BLN/Capstone-Project).

**Comment by Kheirah** [comment]
Kuriöser Fehler. Ich habe keine direkte Lösung dafür, aber du könntest das Problem umgehen, indem du den Export in _BlogTagData.js_ **default** machst und den Import in BlogTag.js entsprechend ohne geschweifte Klammern vornimmst.

**Reply by SLS-BLN** [reply]
Hallo Ernst, danke für deine Antwort - die Idee ist gut, funktioniert aber (leider) auch nicht. Mittlerweile hatte ich auch eine Kurz-Session mit @pixelass und @stefankorder zum Thema - ein paar Dinge konnten wir ausschließen; vermutlich hängt es mit dem Cache zusammen. Die Lösung war dann, die Datei neu anzulegen, so dass die App wieder funktioniert ... 
Viele Grüße,
Stefan
